# This corrects the type of dev_set_mac_address.  The definition (prototype)
# was *added* to include/linux/netdevice.h in the 2.6.11 mm patches, so the
# change should not harm older builds
--- ixp425-eth-1.1-r1/ixp425_eth.c.pre-mm4	2005-03-19 14:02:32.883601440 -0800
+++ ixp425-eth-1.1-r1/ixp425_eth.c	2005-03-19 14:03:25.849549392 -0800
@@ -2347,12 +2347,11 @@
 }
 
 /* set port MAC addr and update the dev struct if successfull */
-int dev_set_mac_address(struct net_device *dev, void *addr)
+int dev_set_mac_address(struct net_device *dev, struct sockaddr *saddr)
 {
     int res;
     priv_data_t *priv = dev->priv;
     IxEthAccMacAddr npeMacAddr;
-    struct sockaddr *saddr = (struct sockaddr *)addr;
 
     /* Set MAC addr in h/w */
     memcpy(&npeMacAddr.macAddress,
